frontend/src/stores/user.js

```diff
@@ -9,113 +15,120 @@ const DEFAULT_PIN = {
     '3': '',
 };
 
-// local utility helper
-const toStringFromObject = (obj) => Object.values(obj).join('');
-const processPhoneNumber = (phoneNumber) =>
-    phoneNumber.replace(/^0/, '+358').replace(/\s/g, '');
-
-class UserStore {
+class userStore {
     isAuthenticating = false;
-    token = null;
     authenticationFailed = false;
-    phoneNumber = '';
-    phoneNumberIncorrect = false;
     pinCode = DEFAULT_PIN;
-    pinCodeIsSet = false;
-    balance = 100;
+    username = undefined;
+    phoneNumber = undefined;
+    token = undefined;
+    balance = 0;
 
+    constructor(rootStore) {
+        this.rootStore = rootStore;
+        try {
+            const userData = hydrateFromStorage('user');
```

Comment:
How are we now checking that the user is actually signed in still? The way I see it, the user details should be fetched every time the app is opened to see that the authentication is valid still and to update the latest information from the server.

Reply:
right now the strategy is to save username, and phoneNumber and token into the system, but this is just place holder.
Maybe on app startup, a login request will be sent to the server?

Reply:
oh but then user would effectively be kicked out anyway because the pincode is completely randomized
I think we have to discuss this

Reply:
So how it works is that the requests either fail or succeed based on whether the signed cookies are set, so the backend is able to detect that on every request. So once the app is opened, an request to query the user data (like `/users/me`) is needed. If this does return 401, the user has not authenticated and login page should be shown. If this returns the data, continue to use the app as normal and user can be expected to be signed in on frontend.